src/EnqueueMessageProcessor.php

```diff
@@ -0,0 +1,109 @@
+<?php
+namespace Prooph\ServiceBus\Message\Enqueue;
+
+use Enqueue\Client\CommandSubscriberInterface;
+use Enqueue\Consumption\QueueSubscriberInterface;
+use Enqueue\Consumption\Result;
+use Enqueue\Psr\PsrContext;
+use Enqueue\Psr\PsrMessage;
+use Enqueue\Psr\PsrProcessor;
+use Enqueue\Util\JSON;
+use Prooph\Common\Messaging\Message;
+use Prooph\ServiceBus\CommandBus;
+use Prooph\ServiceBus\EventBus;
+use Prooph\ServiceBus\QueryBus;
+
+class EnqueueMessageProcessor implements PsrProcessor, QueueSubscriberInterface, CommandSubscriberInterface
+{
+    /**
+     * @var CommandBus
+     */
+    private $commandBus;
+
+    /**
+     * @var EventBus
+     */
+    private $eventBus;
+
+    /**
+     * @var QueryBus
+     */
+    private $queryBus;
+
+    /**
+     * @var EnqueueSerializer
+     */
+    private $serializer;
+
+    /**
+     * @param CommandBus $commandBus
+     * @param EventBus $eventBus
+     * @param QueryBus $queryBus
+     * @param EnqueueSerializer $serializer
+     */
+    public function __construct(CommandBus $commandBus, EventBus $eventBus, QueryBus $queryBus, EnqueueSerializer $serializer)
+    {
+        $this->commandBus = $commandBus;
+        $this->eventBus = $eventBus;
+        $this->queryBus = $queryBus;
+
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function process(PsrMessage $psrMessage, PsrContext $psrContext)
+    {
+        $message = $this->serializer->unserialize($psrMessage->getBody());
+
+        switch ($message->messageType()) {
+            case Message::TYPE_EVENT:
+                $this->eventBus->dispatch($message);
+
+                break;
+            case Message::TYPE_COMMAND:
+                $this->commandBus->dispatch($message);
+
+                break;
+            case Message::TYPE_QUERY:
+                $promise = $this->queryBus->dispatch($message);
+
+                $body = null;
+                $promise->then(function ($value) use (&$body) {
+                    $body = JSON::encode($value);
+                });
+
+                return Result::reply($psrContext->createMessage($body));
+            default:
+                return Result::reject(sprintf(
+                    'The message type "%s" is invalid. The supported types are "%s"',
+                    $message->messageType(),
+                    implode('", "', [Message::TYPE_COMMAND, Message::TYPE_EVENT, Message::TYPE_QUERY])
+                ));
+        }
+
+        return self::ACK;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedCommand()
+    {
+        return [
+            'processorName' => Commands::PROOPH_BUS,
+            'queueName' => Commands::PROOPH_BUS,
+            'queueNameHardcoded' => true,
+            'exclusive' => true,
+        ];
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function getSubscribedQueues()
+    {
+        return Commands::PROOPH_BUS;
```

Comment:
I completely removed those static methods. In the case of SimpleClient (plain PHP) they are not used. In case of Symfony one can add this configuration on a container tag or write its own processor with implemented static methods and use this one internally. 

Sound good?